multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/util/ApplicationConfiguration.java

```diff
@@ -218,13 +225,24 @@ public class ApplicationConfiguration {
     private Integer threadsForFileStorageUpload;
     private Boolean isHealthCheckEnabled;
 
-    public ApplicationConfiguration() {
-        this(new Environment());
+    private final ApplicationConfigurationAuditLog applicationConfigurationAuditLog;
+
+    public ApplicationConfiguration(ApplicationConfigurationAuditLog applicationConfigurationAuditLog) {
+        this(new Environment(), applicationConfigurationAuditLog);
     }
 
     @Inject
-    public ApplicationConfiguration(Environment environment) {
+    public ApplicationConfiguration(Environment environment, ApplicationConfigurationAuditLog applicationConfigurationAuditLog) {
         this.environment = environment;
+        this.applicationConfigurationAuditLog = applicationConfigurationAuditLog;
+    }
+
+    private void auditLog(String envVariableName, String space) {
+        applicationConfigurationAuditLog.logEnvironmentVariableRead(envVariableName, space);
+    }
+
+    public void auditLog(String envVariableName) {
+        applicationConfigurationAuditLog.logEnvironmentVariableRead(envVariableName, getSpaceGuid());
```

Comment:
getSpaceGuid() appears to call 'auditLog(SPACE_GUID, spaceGuid);' > which is the above method and would result in the same log twice here?

Reply:
reread the code - this would be true only for the first time read the spaceGuid, still looks a bit weird; maybe the two methods should be overloaded so the log would only happen once